File: TopInterview150/Hashmap/LongestConsecutiveSequence/solution.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        mySet = set()
        output = 1
        if len(nums) == 0:
            return 0
        for i in nums:
            mySet.add(i)
        for i in mySet:
            if i-1 not in mySet:
                counter = 1
                curr = i
                while curr+1 in mySet:
                    counter+=1
                    curr+=1
                    output = max(output, counter)
        return output

File: TopInterview150/Hashmap/LongestConsecutiveSequence/solution.cpp
class Solution {
public:
    int longestConsecutive(vector<int>& nums) {
        if (nums.size() == 0) return 0;
        unordered_set<int> set;
        int output = 1;
        for (int n : nums) set.insert(n);//add numbers to set
        for (int i : set)
        {
            if (!set.contains(i-1))//start checks only when encountering a new minimum
            {
                int curr = i;
                int counter = 1;
                while (set.contains(curr+1))
                {
                    curr++;
                    counter++;
                }
                output = max(output, counter);
            }
        }
        return output;
    }
};